waku/node/waku_node.nim

```diff
@@ -361,22 +365,50 @@ proc filterHandleMessage*(node: WakuNode, pubsubTopic: PubsubTopic, message: Wak
   await node.wakuFilter.handleMessage(pubsubTopic, message)
   await node.wakuFilterLegacy.handleMessage(pubsubTopic, message) #TODO: remove legacy
 
-
-proc mountFilterClient*(node: WakuNode) {.async, raises: [Defect, LPError].} =
+proc mountFilterClient*(node: WakuNode, filterV2MessagePushHandler= none(filter_client.MessagePushHandler)) {.async, raises: [Defect, LPError].} =
+  ## Mounting both filter clients v1 - legacy and v2.
+  ## Giving option for application level to chose btw own push message handling or
+  ## rely on node provided cache. - This only applies for v2 filter client
   info "mounting filter client"
 
   node.wakuFilterClientLegacy = WakuFilterClientLegacy.new(node.peerManager, node.rng)
+
+  if filterV2MessagePushHandler.isNone():
+    # TODO: tie it to config, but from proper place (rest_legacy_filter_api.filterMessageCacheDefaultCapacity)
+    node.filterV2PushMessageCache = TopicCache.init(capacity=30)
+
+    let internalPushMessageHandler: filter_client.MessagePushHandler = proc(pubsubTopic: string, message: WakuMessage) {.gcsafe, closure.} =
+      node.filterV2PushMessageCache.addMessage(pubsubTopic, message)
+
+    node.wakuFilterClient = filter_client.WakuFilterClient.new(node.rng, internalPushMessageHandler, node.peerManager)
+  else:
+    node.wakuFilterClient = filter_client.WakuFilterClient.new(node.rng, filterV2MessagePushHandler.get(), node.peerManager)
+
   if node.started:
-    # Node has started already. Let's start filter too.
-    await node.wakuFilterClientLegacy.start()
+    await allFutures(node.wakuFilterClientLegacy.start(), node.wakuFilterClient.start())
 
+  node.switch.mount(node.wakuFilterClient, protocolMatcher(WakuFilterSubscribeCodec))
   node.switch.mount(node.wakuFilterClientLegacy, protocolMatcher(WakuFilterCodec))
 
-proc filterSubscribe*(node: WakuNode, pubsubTopic: Option[PubsubTopic], contentTopics: ContentTopic|seq[ContentTopic],
-                handler: FilterPushHandler, peer: RemotePeerInfo|string) {.async, gcsafe, raises: [Defect, ValueError].} =
+# TODO: review this if it is sufficient app interface or not.
+proc getFilterV2PushMessageCache*(node: WakuNode): Option[TopicCache] =
```

Comment:
Fine for now but I'm thinking about adding a `waku_cache` sub-system in the near future.